File: Data Structure I/cpp/Day 14/TwoSumBST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool findTarget(TreeNode* root, int k) {
        vector<int> list;
        has(root,list);
        int s=0;
        int e=list.size()-1;
        while(s<e){
            int sum=list[s]+list[e];
            if(sum==k){
                return true;
            }
            if(sum>k){
                e--;
            }else{
                s++;
            }

        }
        return false;

    }
    void has(TreeNode* root,vector<int>& list){
         if(root==NULL){
             return;
         }
         has(root->left,list);
         list.push_back(root->val);
         has(root->right,list);
    }
};

File: Data Structure I/python/Day 14/TwoSumBST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findTarget(self, root: Optional[TreeNode], k: int) -> bool:
        lst=[]
        self.has(root,lst)
        s=0
        e=len(lst)-1
        while(s<e):
            sum=lst[s]+lst[e]
            if(sum==k):
                return True
            
            if(sum>k):
                e-=1
            else:
                s+=1
        return False

    
    def has(self,root,lst):
         if(root==None):
             return
         
         self.has(root.left,lst)
         lst.append(root.val)
         self.has(root.right,lst)